domain/storage/service/storagepool.go

```diff
@@ -186,11 +191,15 @@ func (s *StoragePoolService) ReplaceStoragePool(ctx context.Context, name string
 
 	// Use the existing provider type unless explicitly overwritten.
 	if providerType == "" {
-		existingConfig, err := s.st.GetStoragePoolByName(ctx, name)
+		spUUID, err := s.st.GetStoragePoolUUID(ctx, name)
```

Comment:
There is probably work to happen outside of this more. But knowing that we will change this func to update on UUID I would suggest just doing this call at the top outside of the if scope.